util/reggen/data.py

```diff
@@ -218,10 +218,12 @@ class Block():
     addr_width = 12
     base_addr = 0
     name = ""
+    instance = ""
```

Comment:
This looks a bit surprising to me. Are you sure this should be a class attribute? It looks like different instances of `Block` should have different `self.instance` values. If so, this code is only working because it turns out that we produce and consume blocks one at a time :-)

Reply:
As discussed with @sriyerg change to use a OrderDict for base_addr and instance name is the key. PTAL

Reply:
I thought the blocks were being sorted anyway. I guess its fine.

Reply:
Ok, looks like this works. I'm pretty certain that the class attributes are unnecessary (and will cause mysterious bugs in the future!), but that's a more general question about the reggen code, not this PR.

Reply:
That's good point. Filed #4341 for further discussion. Thanks